### PR TITLE
quickgui: 1.2.8 -> 1.2.10

### PR DESCRIPTION
**Pull request by flexiondotorg**
## Description of changes

I am one of the upstream developers of Quickgui. This pull request updates Quickgui to 1.2.10 and changes the package to build from source rather than dump a .deb.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>quickgui</li>
    <li>quickgui.debug</li>
    <li>quickgui.pubcache</li>
  </ul>
</details>

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ToasterUwU** [comment]
Anything i can help with?

There is a merge conflict and also a newer version out by now. Happy to test or whatever is needed to get this on the way

**Comment by heyimnova** [comment]
Sorry for taking so long to get to this. The merge conflict is because gnome.zenity has been renamed to zenity since this was forked so this branch needs to be rebased to current master and have these changes reapplied.

File: pkgs/applications/virtualization/quickgui/default.nix [hunk 1]
```diff
@@ -1,67 +1,60 @@
...
-
-    mv usr $out
+  gitHashes = {
+    window_size = "sha256-XelNtp7tpZ91QCEcvewVphNUtgQX7xrp5QP0oFo6DgM=";
+  };
 
-    runHook postInstall
-  '';
+  # These things are added to LD_LIBRARY_PATH, but not PATH
+  runtimeDependencies = [ quickemu ];
```
Comment (heyimnova) [inline pkgs/applications/virtualization/quickgui/default.nix:28]: I don't think quickemu needs to be included here, it builds and works without it being here when it's already included in runtimeBinDependencies.

**Review by heyimnova — CHANGES_REQUESTED** [review]

**Comment by ToasterUwU** [comment]
Also, if at all possible, it would be nice to update this to 1.2.10 if possible, otherwise we will update from old version to less old version, which is suboptimal.

File: pkgs/applications/virtualization/quickgui/default.nix [hunk 2]
```diff
@@ -1,67 +1,60 @@
...
-, wrapGAppsHook3
-, quickemu
+, flutter
 , gnome
+, quickemu
 }:
-
-stdenvNoCC.mkDerivation rec {
+let
+  runtimeBinDependencies = [ gnome.zenity ];
```
[the remark this answers is not in the document]
Comment (flexiondotorg) [reply pkgs/applications/virtualization/quickgui/default.nix]: Rebased and renamed `gnome.zenity` to `zenity`.

File: pkgs/applications/virtualization/quickgui/default.nix (hunk 1 above)
Comment (flexiondotorg) [reply pkgs/applications/virtualization/quickgui/default.nix]: Rebased and made the requested change to include `quickemu` in the `runtimeBinDependencies`.

**Comment by flexiondotorg** [comment]
Updated to 1.2.10 as requested by @ToasterUwU

**Comment by ToasterUwU** [comment]
Result of `nixpkgs-review pr 323381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>quickgui</li>
    <li>quickgui.debug</li>
    <li>quickgui.pubcache</li>
  </ul>
</details>

**Review by ToasterUwU — APPROVED** [review]

**Comment by ToasterUwU** [comment]
Builds fine, starts fine, downloads fine, runs the VM just fine. 

LGTM

File: pkgs/applications/virtualization/quickgui/default.nix [hunk 3]
```diff
@@ -1,67 +1,59 @@
...
-    runHook preInstall
-
-    mv usr $out
+  gitHashes = {
+    window_size = "sha256-XelNtp7tpZ91QCEcvewVphNUtgQX7xrp5QP0oFo6DgM=";
+  };
 
-    runHook postInstall
-  '';
+  # These things are added to LD_LIBRARY_PATH, but not PATH
```
Comment (heyimnova) [inline pkgs/applications/virtualization/quickgui/default.nix:27]: I don't think this comment is accurate here anymore but everything builds and works so the package is fine.

**Review by heyimnova — APPROVED** [review]

File: pkgs/applications/virtualization/quickgui/default.nix (hunk 3 above)
Comment (flexiondotorg) [reply pkgs/applications/virtualization/quickgui/default.nix]: Comment removed.

**Review by ToasterUwU — APPROVED** [review]

**Review by heyimnova — APPROVED** [review]

**Comment by ToasterUwU** [comment]
@heyimnova Since you are the package maintaner in this case, take a look at: https://github.com/nixos/nixpkgs-merge-bot

If this is actually up and running and works like intended, you can merge this PR without external help

**Review by phanirithvij — APPROVED** [review]

**Comment by pbsds** [comment]
It seems `env -i ldd` is complaining about these two files:
```
results/quickgui/app/lib/libwindow_size_plugin.so
	libflutter_linux_gtk.so => not found
results/quickgui/app/lib/liburl_launcher_linux_plugin.so
	libflutter_linux_gtk.so => not found
```

**Comment by flexiondotorg** [comment]
> env -i ldd

All the required files are present when I run `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"` and the resulting executable runs correctly.

File: pkgs/applications/virtualization/quickgui/default.nix [hunk 4]
```diff
@@ -1,67 +1,58 @@
...
+      icon = "quickgui";
+      desktopName = "Quickgui";
+      comment = "An elegant virtual machine manager for the desktop";
+      categories = [ "Development" "System" ];
+    })
+  ];
+
   meta = with lib; {
-    description = "Flutter frontend for quickemu";
+    description = "An elegant virtual machine manager for the desktop";
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/quickgui/default.nix:50]:
```suggestion
    description = "Elegant virtual machine manager for the desktop";
```

File: pkgs/applications/virtualization/quickgui/default.nix [hunk 5]
```diff
@@ -1,67 +1,58 @@
...
-  src = fetchurl {
-    url = "https://github.com/quickemu-project/quickgui/releases/download/v${version}/quickgui_${version}-1_lunar1.0_amd64.deb";
-    sha256 = "sha256-crnV7OWH5UbkMM/TxTIOlXmvqBgjFmQG7RxameMOjH0=";
+  version = "1.2.10";
+  src = fetchFromGitHub {
+    owner = "quickemu-project";
+    repo = "quickgui";
+    rev = version;
+    hash = "sha256-M2Qy66RqsjXg7ZpHwaXCN8qXRIsisnIyaENx3KqmUfQ=";
   };
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/quickgui/default.nix:20]:
```suggestion
  };

```

File: pkgs/applications/virtualization/quickgui/default.nix [hunk 6]
```diff
@@ -1,67 +1,58 @@
...
-, wrapGAppsHook3
+, flutter
 , quickemu
 , zenity
 }:
-
-stdenvNoCC.mkDerivation rec {
+let
+  runtimeBinDependencies = [ quickemu zenity ];
+in
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/quickgui/default.nix:11]:
```suggestion

```

File: pkgs/applications/virtualization/quickgui/default.nix [hunk 7]
```diff
@@ -1,67 +1,58 @@
...
-    runHook preInstall
-
-    mv usr $out
+  gitHashes = {
+    window_size = "sha256-XelNtp7tpZ91QCEcvewVphNUtgQX7xrp5QP0oFo6DgM=";
+  };
 
-    runHook postInstall
-  '';
+  extraWrapProgramArgs = "--prefix PATH : ${lib.makeBinPath runtimeBinDependencies}";
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/quickgui/default.nix:27]:
```suggestion
  extraWrapProgramArgs = "--prefix PATH : ${lib.makeBinPath [ quickemu zenity ]}";
```

**Comment by ToasterUwU** [comment]
@flexiondotorg Please force push every change in a single commit again with the proper formatting of the description ( "quickgui: 1.2.8 -> 1.2.10" )

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 323381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>quickgui</li>
    <li>quickgui.debug</li>
    <li>quickgui.pubcache</li>
  </ul>
</details>

**Review by JohnRTitor — APPROVED** [review]
LGTM, ran the application successfully, will squash merge

**Pull request merged**